test/metabase/models/channel_test.clj

```diff
@@ -39,6 +38,9 @@
       (is (pos? (t2/update! :model/Channel id {:active true})))
       (is (t2/exists? :model/PulseChannel pc-id)))
 
-   (testing "deactivate channel will delete pulse channels"
-     (t2/update! :model/Channel id {:active false})
-     (is (not (t2/exists? :model/PulseChannel pc-id))))))
+    (testing "deactivate channel"
+      (t2/update! :model/Channel id {:active false})
+      (testing "will delete pulse channels"
+        (is (not (t2/exists? :model/PulseChannel pc-id))))
+      (testing "will change the name"
+        (is (= (format "DEACTIVATED_%d New name" id) (t2/select-one-fn :name :model/Channel id)))))))
```

Comment:
Rather than (just) testing the implementation, I think it's worth checking the use case it supports - creating a new channel with the same original name.

Reply:
I have a test for that in the API namespace [here](https://github.com/metabase/metabase/pull/46818/files#diff-55c06ecf148871a20347412ddf41f2754a028295033cccd0e71ae040c482a167R75)